### PR TITLE
Tone alarm

### PR DESCRIPTION
**Pull request by jpaali**
Please use [PX4 Discuss](http://discuss.px4.io/) or [Slack](http://slack.px4.io/) to align on pull requests if necessary. You can then open draft pull requests to get early feedback.

**Describe problem solved by this pull request**
A clear and concise description of the problem this proposed change will solve.
E.g. For this use case I ran into...

**Describe your solution**
A clear and concise description of what you have implemented.

**Describe possible alternatives**
A clear and concise description of alternative solutions or features you've considered.

**Test data / coverage**
How was it tested? What cases were covered? Logs uploaded to https://review.px4.io/ and screenshots of the important plot parts.

**Additional context**
Add any other related context or media.

### CODE REVIEW

File: src/drivers/tone_alarm/ToneAlarm.cpp [hunk 1]
```diff
@@ -51,13 +53,11 @@ ToneAlarm::ToneAlarm() :
 
 ToneAlarm::~ToneAlarm()
 {
-	ToneAlarmInterface::stop_note();
+	note_must_stop = true;
```
Comment (jlaitine) [inline src/drivers/tone_alarm/ToneAlarm.cpp:56]: This won't work, since ::Run is not any more executed after the destruction...

**Comment by jlaitine** [comment]
Looks good, just destruction case needs to be checked; it won't make much difference for us but for upstreaming that part needs to be checked - how can you make the tone stop when destructing the object

File: boards/ssrc/saluki/init/rc.board_defaults [hunk 2]
```diff
@@ -10,5 +10,6 @@ pwm_esc start
 
 param set-default COM_PREARM_MODE 1
 param set-default CBRK_IO_SAFETY 0
+param set-default CBRK_BUZZER 0
```
[the remark this answers is not in the document]
Comment (jpaali) [reply boards/ssrc/saluki/init/rc.board_defaults:13]: ok

File: src/drivers/tone_alarm/ToneAlarm.h [hunk 3]
```diff
@@ -86,4 +88,6 @@ class ToneAlarm : public ModuleBase<ToneAlarm>, public px4::ScheduledWorkItem
 
 	bool _circuit_break_initialized{false};
 	bool _play_tone{false};
+	bool _initialized{false};
+	static bool note_must_stop;
```
[the remark this answers is not in the document]
Comment (jpaali) [reply src/drivers/tone_alarm/ToneAlarm.h:92]: _stop_tone is used in static member function InterruptStopNote(). Therefore it is specified as static.

**Pull request merged**